### PR TITLE
Adding processor for PFOAnalysis on TrueJet jets

### PR DESCRIPTION
**Pull request by beyerja**
BEGINRELEASENOTES
- Adding new analysis tool: Combined the PFOAnalysis processor with the jet analysis power of the TrueJet/TrueJet_Parser tools to gain insight into individual jet behaviour and reconstruction.

ENDRELEASENOTES

### CODE REVIEW

**Comment by petricm** [comment]
I would be appreciated if you can look at the compile warnings from `TJjetsPFOAnalysisProcessor` and fix them.

**Comment by beyerja** [comment]
The code should now compile without any warnings from my processor - thank you for the comment!
@petricm To achieve this I had to make some changes in the TrueJet_Parser header. Can you make sure that these changes do not influence the functionality of the TrueJet_Parser?

**Comment by petricm** [comment]
Thank you for the fixes, I think everything is fine, I would however have just one more comment. When you edited `TrueJet_Parser` you did quite some white space changes that are not necessary, probably as a result of auto formatting (e.g. white space at the end of line), please remove them.

**Comment by beyerja** [comment]
I had added another feature and I've re-added the whitespaces, hope it is fine now.

File: Analysis/TJjetsPFOAnalysisProcessor/src/FindTrueJetParticles.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#include "TJjetsPFOAnalysisProcessor.h"
+
+void TJjetsPFOAnalysisProcessor::findTrueJetParticles(LCEvent* event) {
+  for (int i_jet=0; i_jet<njets(); i_jet++) {
+    ReconstructedParticleVec jet_recos = seen_partics(i_jet);
+
+    LCRelationNavigator* relation_TrueJetFinalColourNeutral   =  new LCRelationNavigator( event->getCollection( _finalColourNeutralLink ) );
```
Comment (rete) [inline Analysis/TJjetsPFOAnalysisProcessor/src/FindTrueJetParticles.cpp]: Memory leak here. You must delete them at the end of the method

File: Analysis/TJjetsPFOAnalysisProcessor/src/FindTrueJetParticles.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        MCParticle* jet_mc = dynamic_cast<MCParticle*>(jet_mc_objects[i_mc]);
+        streamlog_out(DEBUG) << " FinalCN " << jet_mc << std::endl;
+        if ( jet_mc != NULL ) {
+          unique_jet_mcs.insert( jet_mc );
+        }
+      }
+    }
+    MCParticleVec jet_mcs (unique_jet_mcs.begin(), unique_jet_mcs.end());
+
+    m_jets.push_back( new JetContentPair(jet_mcs, jet_recos) );
```
Comment (rete) [inline Analysis/TJjetsPFOAnalysisProcessor/src/FindTrueJetParticles.cpp]: The objects inside are not deleted anywhere. You have memory leak here !

**Pull request merged**